From edc8b646aa2b7ce86c0b66d69597aafd65963498 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 25 May 2021 10:19:20 +0300
Subject: [PATCH 09/10] clocksource: pit: Correct IRQ affinity

commit b391c3ffc414ec4d6537aa2fbad802bb4fae5cc6 from
https://source.codeaurora.org/external/autobsps32/linux

Replace IRQF_IRQPOLL with IRQF_PERCPU. This will avoid the
schedule of PIT interrupts on cores that are not assigned
to this module.

Upstream-Status: pending

Issue: ALB-6311
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/clocksource/timer-vf-pit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clocksource/timer-vf-pit.c b/drivers/clocksource/timer-vf-pit.c
index 34740ecc72f8..f0bcdb07d652 100644
--- a/drivers/clocksource/timer-vf-pit.c
+++ b/drivers/clocksource/timer-vf-pit.c
@@ -270,7 +270,7 @@ static int pit_clockevent_init(struct pit_timer *pit,
 	pit->clockevent_pit.irq = irq;
 
 	pit->pit_timer_irq.name = TIMER_NAME;
-	pit->pit_timer_irq.flags = IRQF_TIMER | IRQF_IRQPOLL;
+	pit->pit_timer_irq.flags = IRQF_TIMER | IRQF_PERCPU;
 	pit->pit_timer_irq.handler = pit_timer_interrupt;
 	pit->pit_timer_irq.dev_id = &pit->clockevent_pit;
 
-- 
2.25.1

